src/edn/mod.rs

```diff
@@ -165,69 +167,72 @@ impl Map {
 
 impl core::fmt::Display for Vector {
     fn fmt(&self, f: &mut core::fmt::Formatter<'_>) -> core::fmt::Result {
-        write!(
-            f,
-            "[{}]",
-            self.0
-                .iter()
-                .map(ToString::to_string)
-                .fold(String::new(), |mut acc, i| {
-                    acc.push_str(&i);
-                    acc.push_str(", ");
-                    acc
-                })
-        )
+        write!(f, "[")?;
+        let mut it = self.0.iter().peekable();
```

Comment:
Have you checked the impact of making everything in the library peekable? Might be a good idea

Reply:
I would guess this is faster because there's no need to allocate a new string.

Reply:
@Grinkers ?